build_tools/benchmarks/benchmark_helper.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env python3
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""Miscellaneous tool to help work with benchmark suite and benchmark CI."""
+
+import pathlib
+import sys
+
+# Add build_tools python dir to the search path.
+sys.path.insert(0, str(pathlib.Path(__file__).parent.with_name("python")))
+
+import argparse
+import json
+from typing import List
+
+from e2e_test_framework import serialization
+from e2e_test_artifacts import model_artifacts, iree_artifacts
+from e2e_test_framework.definitions import iree_definitions
+
+IREE_COMPILER_NAME = "iree-compile"
+
+
+def _dump_cmds_of_generation_config(
+    gen_config: iree_definitions.ModuleGenerationConfig,
+    root_path: pathlib.PurePath = pathlib.PurePath()):
+
+  imported_model = gen_config.imported_model
+  imported_model_path = iree_artifacts.get_imported_model_path(
+      imported_model=imported_model, root_path=root_path)
+  module_path = iree_artifacts.get_module_dir_path(
+      module_generation_config=gen_config,
+      root_path=root_path) / iree_artifacts.MODULE_FILENAME
+  compile_cmds = [
+      IREE_COMPILER_NAME,
+      str(imported_model_path), "-o",
+      str(module_path)
+  ]
+  compile_cmds += gen_config.materialize_compile_flags()
+
+  if imported_model.import_config.tool == iree_definitions.ImportTool.NONE:
+    import_cmds = ["(Source model is already in MLIR)"]
+  else:
+    source_model_path = model_artifacts.get_model_path(
+        model=imported_model.model, root_path=root_path)
+    import_cmds = [
+        imported_model.import_config.tool.value,
+        str(source_model_path), "-o",
+        str(imported_model_path)
+    ]
+    import_cmds += imported_model.import_config.materialize_import_flags(
+        model=imported_model.model)
+
+  # TODO(#12215): Print benchmark name to make them searchable by keywords.
+  # Insert a blank line after each command to help read with line wrap.
+  return [
+      f'Compile Module: {" ".join(compile_cmds)}', "",
```

Comment:
You might want to escape/quote the commands here. [`shlex.quote`](https://docs.python.org/3/library/shlex.html#shlex.quote) will do it for Unix (can't use `shlex.join` until 3.8). I'm not sure if it makes things *worse* for non-Unix. I think the subprocess undocumented API list2cmdline might be relatively cross platform.

Reply:
list2cmdline seems to be only for Windows. Added a condition to use list2cmdline for Windows and shlex.quote for others. Tested on Windows (and seems to work)